src/debugger/gdb-stub.c

```diff
@@ -433,7 +453,35 @@ static void _processQSupportedCommand(struct GDBStub* stub, const char* message)
 		}
 		message = end + 1;
 	}
-	strncpy(stub->outgoing, "swbreak+;hwbreak+", GDB_STUB_MAX_LINE - 4);
+	strncpy(stub->outgoing, "swbreak+;hwbreak+;qXfer:features:read+;qXfer:memory-map:read+", GDB_STUB_MAX_LINE - 4);
+}
+
+static void _processQXferCommand(struct GDBStub* stub, const char* params, const char* data) {
+	unsigned int offset = 0;
+	unsigned int length = 0;
+
+	unsigned int index = 0;
+	while (params[index] != ',') {
+		offset <<= 4;
+		offset |= _hex2int(&params[index], 1);
+		index++;
+	}
+
+	index++;
+	while (index < strlen(params) - 3) {
+		length <<= 4;
+		length |= _hex2int(&params[index], 1);
+		index++;
+	}
+
+	memset(stub->outgoing, 0, GDB_STUB_MAX_LINE);
+	if (strlen(data) - offset < length) {
```

Comment:
These values are unsigned and thus vulnerable to underflow. Also what is this check for? Telling GDB it's reached the end of the value?

```suggestion
	if (strlen(data) < length + offset) {
```

Reply:
Sending an l first means it's the last packet, an m means there is more data after, GDB handles the logic of what offset and length to send.

this check is to know if we are about to read until the end which would change the first character to send and the amount of bytes to copy.